Japanese input no longer works on 17.09 beta whereas it did on 17.03 stable

## Issue description
Japanese input no longer works on 17.09 beta whereas it did on 17.03 stable.

### Steps to reproduce
Add the following to your nixos configuration file:

````
i18n.inputMethod = {
  enabled = "ibus";
  ibus.engines = with pkgs.ibus-engines; [ anthy ];
};
````
`sudo nixos-rebuild switch`

Note the output:

````
aaron@moya /data> sudo nixos-rebuild switch
building Nix...
building the system configuration...
these derivations will be built:
  /nix/store/wabxkjdfrk5ljfyffwwpgkb0q1gwljs1-ibus-anthy-1.5.9.drv
  /nix/store/mdlxrb5s9albgy5rl12gb9z5hv007p9n-ibus-with-plugins-1.5.16.drv
  /nix/store/5l57kfbkjh306m7h4dwjg7rz8cyyd435-gtk3-immodule.cache.drv
  /nix/store/74c5v88ya73djlm8n9jgh4fkfrc8a545-autostart-ibus-daemon.drv
  /nix/store/qpxxl3i2sz4k1kbr74gi48j4s92chcla-gtk2-immodule.cache.drv
  /nix/store/s8i9nmwhn10xv5bnvf7n0q1k79g08kcz-system-path.drv
  /nix/store/ggi27b2zai9n0bd719kljr37jpvm7a8r-dbus-1.drv
  /nix/store/1qf4vixkgds2bccc8j6d7mzw55x5l1xx-unit-dbus.service.drv
  /nix/store/50p1n2wny6821c5pggpl7hj5b0ln7z8g-unit-polkit.service.drv
  /nix/store/xwy1515pbk722mnx362ndjlk53g4lwz5-unit-systemd-fsck-.service.drv
  /nix/store/m2x9x3vqm8nis04rzg43hv1m07wjav37-system-units.drv
  /nix/store/ziz1zds1pjss55gvn9fpd1hgs55gxvf8-user-units.drv
  /nix/store/x03dlfc36xlyf0aymabm4rmksfgv6y9z-etc.drv
  /nix/store/2mw628xr453yhlj24gq4ijaask4nrd4x-nixos-system-moya-17.09beta487.773268eede.drv
building path(s) ‘/nix/store/ld0zlnqxhpwz7q6xdn0f85sc0m7vgwyc-ibus-anthy-1.5.9’
unpacking sources
unpacking source archive /nix/store/rz2h6h6r0d28wanapafpg62f3gwww5br-ibus-anthy-1.5.9.tar.gz
source root is ibus-anthy-1.5.9
setting SOURCE_DATE_EPOCH to timestamp 1476933248 of file ibus-anthy-1.5.9/setup/python3/ibus-setup-anthy.in
patching sources
configuring
fixing libtool script ./ltmain.sh
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/ld0zlnqxhpwz7q6xdn0f85sc0m7vgwyc-ibus-anthy-1.5.9 --with-anthy-zipcode=/nix/store/dkbkafa14jxf36clhn9j3w17iq12qqwr-anthy-9100h/share/anthy/zipcode.t
checking for a BSD-compatible install... /nix/store/g6bzbxan7z1s2rfkhz12fcnfy1dmqmhh-coreutils-8.28/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /nix/store/g6bzbxan7z1s2rfkhz12fcnfy1dmqmhh-coreutils-8.28/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking for style of include used by make... GNU
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... none
checking how to run the C preprocessor... gcc -E
checking for grep that handles long lines and -e... /nix/store/jshi9bff0ha6scxkndhl1i6kbmmfxxxz-gnugrep-3.1/bin/grep
checking for egrep... /nix/store/jshi9bff0ha6scxkndhl1i6kbmmfxxxz-gnugrep-3.1/bin/grep -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking minix/config.h usability... no
checking minix/config.h presence... no
checking for minix/config.h... no
checking whether it is safe to define __EXTENSIONS__... yes
checking nano version... 0 (release)
checking whether build environment is sane... yes
checking whether to enable maintainer-specific portions of Makefiles... yes
checking for gcc... (cached) gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ISO C89... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking dependency style of gcc... (cached) none
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... none
checking for ANSI C header files... (cached) yes
checking build system type... x86_64-unknown-linux-gnu
checking host system type... x86_64-unknown-linux-gnu
checking how to print strings... printf
checking for a sed that does not truncate output... /nix/store/03g1ack3mbfbcjd6szi048p5y4x8zarm-gnused-4.4/bin/sed
checking for fgrep... /nix/store/jshi9bff0ha6scxkndhl1i6kbmmfxxxz-gnugrep-3.1/bin/grep -F
checking for ld used by gcc... ld
checking if the linker (ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... nm
checking the name lister (nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-unknown-linux-gnu file names to x86_64-unknown-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-unknown-linux-gnu file names to toolchain format... func_convert_file_noop
checking for ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse nm output from gcc object... ok
checking for sysroot... no
checking for a working dd... /nix/store/g6bzbxan7z1s2rfkhz12fcnfy1dmqmhh-coreutils-8.28/bin/dd
checking how to truncate binary pipes... /nix/store/g6bzbxan7z1s2rfkhz12fcnfy1dmqmhh-coreutils-8.28/bin/dd bs=4096 count=1
./configure: line 8507: /usr/bin/file: No such file or directory
checking for mt... no
checking if : is a manifest tool... no
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (ld) supports shared libraries... yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... ld
checking if the linker (ld) is GNU ld... yes
checking whether the g++ linker (ld) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC -DPIC
checking if g++ PIC flag -fPIC -DPIC works... yes
checking if g++ static flag -static works... no
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (ld) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether NLS is requested... yes
checking for intltool >= 0.41.1... 0.51.0 found
checking for intltool-update... /nix/store/dhmcgknqli5zzkg5l60ahy4fxi6m5lhl-intltool-0.51.0/bin/intltool-update
checking for intltool-merge... /nix/store/dhmcgknqli5zzkg5l60ahy4fxi6m5lhl-intltool-0.51.0/bin/intltool-merge
checking for intltool-extract... /nix/store/dhmcgknqli5zzkg5l60ahy4fxi6m5lhl-intltool-0.51.0/bin/intltool-extract
checking for xgettext... /nix/store/yrkgzgn6cc1vpd7xdds5ng98nvlwaly8-gettext-0.19.8/bin/xgettext
checking for msgmerge... /nix/store/yrkgzgn6cc1vpd7xdds5ng98nvlwaly8-gettext-0.19.8/bin/msgmerge
checking for msgfmt... /nix/store/yrkgzgn6cc1vpd7xdds5ng98nvlwaly8-gettext-0.19.8/bin/msgfmt
checking for gmsgfmt... /nix/store/yrkgzgn6cc1vpd7xdds5ng98nvlwaly8-gettext-0.19.8/bin/msgfmt
checking for perl... /nix/store/pzlhf9j12zlxg2p9n543hk1kvhj363wa-perl-5.24.2/bin/perl
checking for perl >= 5.8.1... 5.24.2
checking for XML::Parser... ok
checking for pkg-config... /nix/store/l0n7c6dbmsiy2bdzqj40yz5p5ih61xri-pkg-config-0.29.2/bin/pkg-config
checking pkg-config is at least version 0.9.0... yes
checking for ANTHY... yes
checking whether NTH_UNCONVERTED_CANDIDATE is declared... yes
checking whether NTH_KATAKANA_CANDIDATE is declared... yes
checking whether NTH_HIRAGANA_CANDIDATE is declared... yes
checking whether NTH_HALFKANA_CANDIDATE is declared... yes
checking for GOBJECT2... yes
checking for gobject-introspection... yes
checking for env... /nix/store/g6bzbxan7z1s2rfkhz12fcnfy1dmqmhh-coreutils-8.28/bin/env
checking for a Python interpreter with version >= 2.5... python
checking for python... /nix/store/kxk4x00irdlfj6bzk58jgx40r76bjzbi-python3-3.6.2/bin/python
checking for python version... 3.6
checking for python platform... linux
checking for python script directory... ${prefix}/lib/python3.6/site-packages
checking for python extension module directory... ${exec_prefix}/lib/python3.6/site-packages
checking if your ibus version is supported... Traceback (most recent call last):
  File "<string>", line 2, in <module>
ModuleNotFoundError: No module named 'gi'
configure: error: no
builder for ‘/nix/store/wabxkjdfrk5ljfyffwwpgkb0q1gwljs1-ibus-anthy-1.5.9.drv’ failed with exit code 1
cannot build derivation ‘/nix/store/mdlxrb5s9albgy5rl12gb9z5hv007p9n-ibus-with-plugins-1.5.16.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/s8i9nmwhn10xv5bnvf7n0q1k79g08kcz-system-path.drv’: 1 dependencies couldn't be built
cannot build derivation ‘/nix/store/2mw628xr453yhlj24gq4ijaask4nrd4x-nixos-system-moya-17.09beta487.773268eede.drv’: 1 dependencies couldn't be built
error: build of ‘/nix/store/2mw628xr453yhlj24gq4ijaask4nrd4x-nixos-system-moya-17.09beta487.773268eede.drv’ failed
````

## Technical details

* System: 17.09beta487.773268eede (Hummingbird)
* Nix version: nix-env (Nix) 1.11.14
* Nixpkgs version: "17.09beta487.773268eede"
* Sandboxing enabled: build-use-sandbox = false

--- a/pkgs/tools/inputmethods/ibus-engines/ibus-anthy/default.nix
+++ b/pkgs/tools/inputmethods/ibus-engines/ibus-anthy/default.nix
@@ -18,7 +18,7 @@ stdenv.mkDerivation rec {
   configureFlags = [ "--with-anthy-zipcode=${anthy}/share/anthy/zipcode.t" ];
 
   buildInputs = [
-    anthy glib gobjectIntrospection gtk3 ibus python3
+    anthy glib gobjectIntrospection gtk3 ibus (python3.withPackages (ps: [ps.pygobject3]))
   ];
 
   nativeBuildInputs = [ intltool pkgconfig python3.pkgs.wrapPython ];

